webpack/scenes/ContentViews/Details/Repositories/ContentViewRepositories.js

```diff
@@ -109,6 +180,9 @@ const ContentViewRepositories = ({ cvId }) => {
 
   useEffect(() => {
     dispatch(getRepositoryTypes());
+    if (Object.keys(details).length === 0 && detailStatus !== STATUS.RESOLVED) {
+      dispatch(getContentViewDetails(cvId));
```

Comment:
Had to add this for tests and to render ContentViewRepositories directly as an independent component.

Reply:
I'd like to look into this more, generally I'd like to avoid adding something just for testing, I can help think about possible solutions.

Reply:
Finally was able to look at this, you can add a default value of a blank array for repositoryIds in that section. You can also pass an  `initialApiState` value to the `renderOptions` object argument to the `renderWithRedux` function. This will allow you to pass in more repository ids that will show initially.

This was my attempt at fixing things, feel free to use anything here:

```diff
diff --git a/webpack/scenes/ContentViews/Details/Repositories/ContentViewRepositories.js b/webpack/scenes/ContentViews/Details/Repositories/ContentViewRepositories.js
index d519036..6a3f9b0 100644
--- a/webpack/scenes/ContentViews/Details/Repositories/ContentViewRepositories.js
+++ b/webpack/scenes/ContentViews/Details/Repositories/ContentViewRepositories.js
@@ -80,7 +80,7 @@ const ContentViewRepositories = ({ cvId }) => {
     setBulkActionOpen(!bulkActionOpen);
   };
   const onAdd = (repos) => {
-    const { repository_ids: repositoryIds } = details;
+    const { repository_ids: repositoryIds=[] } = details;
     dispatch(updateContentView(cvId, { repository_ids: repositoryIds.concat(repos) }));
   };
 
@@ -172,9 +172,6 @@ const ContentViewRepositories = ({ cvId }) => {
 
   useEffect(() => {
     dispatch(getRepositoryTypes());
-    if (Object.keys(details).length === 0 && detailStatus !== STATUS.RESOLVED) {
-      dispatch(getContentViewDetails(cvId));
-    }
   }, []);
 
   // Get repo type filter selections dynamically from the API
diff --git a/webpack/scenes/ContentViews/Details/Repositories/__tests__/contentViewAddRemove.test.js b/webpack/scenes/ContentViews/Details/Repositories/__tests__/contentViewAddRemove.test.js
index 8f9109d..729f608 100644
--- a/webpack/scenes/ContentViews/Details/Repositories/__tests__/contentViewAddRemove.test.js
+++ b/webpack/scenes/ContentViews/Details/Repositories/__tests__/contentViewAddRemove.test.js
@@ -15,7 +15,7 @@ const repoTypesResponse = [{ name: 'deb' }, { name: 'docker' }, { name: 'file' }
 const cvDetailData = require('../../__tests__/contentViewDetails.fixtures.json');
 
 const cvDetailsPath = api.getApiUrl('/content_views/1');
-const cvAddparams = { repository_ids: [58, 62, 64, 107, 106] };
+const cvAddparams = { repository_ids: [107, 106] };
 const cvRepoAddData = require('./contentViewRepoAdd.fixture.json');
 
 let firstRepo;
@@ -32,12 +32,7 @@ beforeEach(() => {
     .get(api.getApiUrl('/repositories/repository_types'))
     .query(true)
     .reply(200, repoTypesResponse);
-  nockInstance
-    .persist()
-    .get(cvDetailsPath)
-    .query(true)
-    .reply(200, cvDetailData);
-});
+ });
 
 afterEach(() => {
   nock.cleanAll();
@@ -79,6 +74,11 @@ test('Can add repositories', async (done) => {
     .put(cvDetailsPath, cvAddparams)
     .reply(200, cvRepoAddData);
 
+  const cvDetailScope = nockInstance
+    .get(cvDetailsPath)
+    .query(true)
+    .reply(200, cvDetailData);
+
   const scope = nockInstance
     .get(cvAllRepos)
     .query(true)
@@ -105,5 +105,6 @@ test('Can add repositories', async (done) => {
   });
   assertNockRequest(autocompleteScope);
   assertNockRequest(repoAddscope);
+  assertNockRequest(cvDetailScope);
   assertNockRequest(scope, done);
 });
diff --git a/webpack/scenes/ContentViews/Details/Repositories/__tests__/contentViewDetailRepos.test.js b/webpack/scenes/ContentViews/Details/Repositories/__tests__/contentViewDetailRepos.test.js
index b32310d..ebfc146 100644
--- a/webpack/scenes/ContentViews/Details/Repositories/__tests__/contentViewDetailRepos.test.js
+++ b/webpack/scenes/ContentViews/Details/Repositories/__tests__/contentViewDetailRepos.test.js
@@ -31,11 +31,6 @@ beforeEach(() => {
     .get(api.getApiUrl('/repositories/repository_types'))
     .query(true)
     .reply(200, repoTypesResponse);
-  nockInstance
-    .persist()
-    .get(cvDetailsPath)
-    .query(true)
-    .reply(200, cvDetailData);
 });
 
 afterEach(() => {
```

Reply:
@sjha4 were you able to look at this?